pkg/controller/endpointslice/utils.go

```diff
@@ -267,6 +270,55 @@ func serviceControllerKey(endpointSlice *discovery.EndpointSlice) (string, error
 	return fmt.Sprintf("%s/%s", endpointSlice.Namespace, serviceName), nil
 }
 
+// setEndpointSliceLabels returns a map with the new endpoint slices labels and true if there was an update
+// The well-known labels IsHeadlessService, LabelServiceName and LabelManagedBy does not generate updates
+// The slice labels must have:
+// 1. the well-known labels discovery.LabelServiceName and discovery.LabelManagedBy
+// 2. the parent service labels
+// 3. the v1.IsHeadlessService label if it is a headless service
```

Comment:
I think with the latest iteration it's actually simpler than this. Something like:

> EndpointSlice labels must be equivalent to the Service labels except for the reserved IsHeadlessService, LabelServiceName and LabelManagedBy labels.